### ISSUE TITLE
Fix false positives for maps in color-no-invalid-hex and property-no-unknown

### ISSUE THREAD

**Issue by strigefleur** [stylelint/stylelint#5654]
Less feature known as maps supposed to be supported atm. However both maps init and maps usage produce false positives.

> Which rule, if any, is the bug related to?

`property-no-unknown`
`color-no-invalid-hex`

> What code is needed to reproduce the bug?

```less
#colors() {
  somecolor: red;
  someothercolor: #fff;
}

.myclass {
  color: #colors[somecolor];
}
```

> What Stylelint configuration is needed to reproduce the bug?

reproducable even without it, but if anything..
had to use `postcss-less` following `stylelint` v.14 update logic where there's no more in-built analyzers.

```json
{
  "extends": "stylelint-config-recommended-less",
  "customSyntax": "postcss-less"
}
```

> Which version of Stylelint are you using?

`14.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Stylelint extension @ VSCode, Stylelint as default less linter

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to LESS maps.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
error Unexpected invalid hex color "#colors" (color-no-invalid-hex)
error Unexpected unknown property "somecolor" (property-no-unknown)
error Unexpected unknown property "someothercolor" (property-no-unknown)
```

Related to [this issue](https://github.com/stylelint/stylelint/issues/4834)
